### PR TITLE
new categories format, and other features

### PR DESCRIPTION
**Pull request by MusiikoAS**

### CODE REVIEW

**Review by susslik — CHANGES_REQUESTED** [review]
Hey @MusiikoAS - just checked the PR, nice job!
However, I found a few possible improvements and added the comments.
And also the PR destination branch is wrong: It's master instead of develop

File: ios/InBrainSurveys.m [hunk 1]
```diff
@@ -14,6 +14,133 @@ - (instancetype)init
     return self;
 }
 
+- (NSString *)profileMatchTitle:(SurveyProfileMatch) profileMatch {
+    switch (profileMatch) {
+        case SurveyProfileMatchNewSurvey:
+            return @"SurveyProfileMatchNewSurvey";
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply ios/InBrainSurveys.m]: done

File: ios/InBrainSurveys.m [hunk 2]
```diff
@@ -14,6 +14,133 @@ - (instancetype)init
...
+    return @"Unknown";
+}
+
+
+- (NSString *)categoryTitle:(InBrainSurveyCategory) category {
+  switch (category) {
+      case InBrainSurveyCategoryAutomotive:
+          return @"InBrainSurveyCategoryAutomotive";
+      case InBrainSurveyCategoryBeveragesAlcoholic:
+          return @"InBrainSurveyCategoryBeveragesAlcoholic";
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply ios/InBrainSurveys.m]: done

File: ios/InBrainSurveys.m [hunk 3]
```diff
@@ -176,20 +303,30 @@ + (BOOL)requiresMainQueueSetup
 {
     @try {
       InBrainSurveyFilter *filterObj = [[InBrainSurveyFilter alloc] initWithPlacementId:placementId categoryIDs:categoryIDs excludedCategoryIDs:excludedCategoryIDs];
-      
-      
+
       [[InBrain shared] getNativeSurveysWithFilter:filterObj success:^
         (NSArray<InBrainNativeSurvey *> * surveys){
               NSMutableArray *surveyList = [NSMutableArray array];
               for(int i = 0; i < surveys.count; i++) {
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply ios/InBrainSurveys.m:248]: done

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 4]
```diff
@@ -265,13 +267,34 @@ public void nativeSurveysReceived(List<Survey> surveys) {
                         map.putBoolean("currencySale", survey.currencySale);
                         map.putDouble("multiplier", survey.multiplier);
 
-
                         WritableArray categories = Arguments.createArray();
+                        WritableArray categoriesNamed = Arguments.createArray();
+                        WritableMap categoryNamed = Arguments.createMap();
                         for (SurveyCategory category:survey.categories) {
+                            categoryNamed.putInt("id", category.getId());
+                            categoryNamed.putString("name", category.name());
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java]: created new separated functions for categories and profileMatch maping

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 5]
```diff
@@ -265,13 +267,34 @@ public void nativeSurveysReceived(List<Survey> surveys) {
...
                             categories.pushInt(category.getId());
-
                         }
-                        map.putArray("categories", categories);
+
+                        map.putArray("categoriesId", categories);
+                        map.putArray("categories", categoriesNamed);
+
+                        //TMP map FOR MAPPING profileMatches
+                        WritableMap matches = Arguments.createMap();
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply android/src/main/java/com/inbrain/InBrainSurveysModule.java]: created new separated functions for categories and profileMatch maping

File: ios/InBrainSurveys.m [hunk 6]
```diff
@@ -14,6 +14,133 @@ - (instancetype)init
...
+}
+
+
+- (NSString *)categoryTitle:(InBrainSurveyCategory) category {
+  switch (category) {
+      case InBrainSurveyCategoryAutomotive:
+          return @"InBrainSurveyCategoryAutomotive";
+      case InBrainSurveyCategoryBeveragesAlcoholic:
+          return @"InBrainSurveyCategoryBeveragesAlcoholic";
+          break;
```
[the remark this answers is not in the document]
Comment (MusiikoAS) [reply ios/InBrainSurveys.m]: done

File: android/src/main/java/com/inbrain/InBrainSurveysModule.java [hunk 7]
```diff
@@ -265,13 +266,24 @@ public void nativeSurveysReceived(List<Survey> surveys) {
                         map.putBoolean("currencySale", survey.currencySale);
                         map.putDouble("multiplier", survey.multiplier);
 
-
                         WritableArray categories = Arguments.createArray();
+                        WritableArray namedCategories = Arguments.createArray();
+                        WritableMap categoryNamed = Arguments.createMap();
```
Comment (susslik) [inline android/src/main/java/com/inbrain/InBrainSurveysModule.java:271]: If I understood correctly - the "id" and "name" values will be re-written at each "for" interaction. And that variable is used only inside "for" loop - so probably we can move it's declaration inside the loop? Just to keep the local variable only where it needed

**Review by susslik — CHANGES_REQUESTED** [review]
Hey @MusiikoAS - the updates look really great! One small comment was added, could you check it pls when you will have a minute?

**Review by susslik — APPROVED** [review]

**Pull request merged**